### ISSUE TITLE
[Windows] Unpooling of nn.FractionalMaxPool2d does not work

### ISSUE THREAD

**Issue by sanctifier** [pytorch/pytorch#12516]
In official online document, the nn.MaxUnpool2d() is recommended to  unpool nn.FractionalMaxPool2d(), however, there are no stride, padding arguments involved in class nn.FractionalMaxPool2d, but the initialization of class nn.MaxUnpool2d does require stride and padding and these parameters affect the shape of the resulting data. 

Thus I wrote a script to compute stride and padding for nn.MaxUnpool2d to unpool nn.FractionalMaxPool2d. This script works fine in Pytorch 0.4, but not in Pytorch 0.4.1. The reason is the none-integer stride and padding trigger exception in forward() of nn.MaxUnpool2d.

## To Reproduce

    import torch, math
    import torch.nn as nn
    import torch.nn.functional as F

    def compute_processed_size(data_size, kernel_size, process_mode='conv', stride=1, padding=0, dilation=1, output_ratio=None):
        if process_mode == 'unpool':
            return math.floor((data_size - 1) * stride - 2 * padding + kernel_size)
    
    size = (100,80)    # sample size
    input = torch.arange(1, size[0]*size[1] + 1).reshape((1,1,size[0],size[1])).float()
    kernel = (2, 2)   
    output_ratio = 0.62   
    kernel, stride, padding = (2,2), (1/output_ratio,1/output_ratio), (0., 0.)
    pool = nn.FractionalMaxPool2d(kernel_size=kernel,output_ratio=output_ratio,return_indices=True)
    output, indices = pool(input)    
    diff0, diff1 = 1, 1
    while diff0 != 0 or diff1 != 0:        
        unpool_size0 = compute_processed_size(output.shape[2], kernel[0],'unpool',stride=stride[0],padding=padding[0])
        unpool_size1 = compute_processed_size(output.shape[3], kernel[1],'unpool',stride=stride[1],padding=padding[1])
        diff0, diff1 = unpool_size0-input.shape[2], unpool_size1 - input.shape[3]
        if diff0 != 0 or diff1 != 0:
            padding= (diff0/2, diff1/2)
    print(f'unpool size {unpool_size0}-by-{unpool_size1}')
    unpool = nn.MaxUnpool2d(kernel_size=kernel,stride=stride,padding=padding)
    unpooled = unpool(output, indices)

The last line throws an exception as followings.
Exception has occurred: TypeError
'float' object cannot be interpreted as an integer
  File "C:\Users\Administrator\Desktop\temp_python\test.py", line 25, in <module>
    unpooled = unpool(output, indices)    

## Expected behavior

nn.MaxUnpool2d can accept none-integer stride and padding parameters as in Pytorch 0.4, OR nn.MaxUnpool2d can unpool result from nn.FractionalMaxPool2d without requiring stride and padding.

## Environment

PyTorch or Caffe2: PyTorch
How you installed PyTorch (conda, pip, source): "conda install pytorch cuda91"
Build command you used (if compiling from source):
OS: Windows 10 Home Edition
PyTorch version: 0.4.1
Python version: 3.6.5 |Anaconda, Inc.| (default, Mar 29 2018, 13:32:41) [MSC v.1900 64 bit (AMD64)]
CUDA/cuDNN version: 9.1.85/7005
GPU models and configuration: GPU 0 = GeForce GTX 1060
GCC version (if compiling from source):
CMake version:
Versions of any other relevant libraries: torchvision (0.2.1)

**Comment by ngimel** [comment]
Closing due to age.
